pkg/porter/list.go

```diff
@@ -13,6 +13,8 @@ import (
 	"get.porter.sh/porter/pkg/storage"
 	"get.porter.sh/porter/pkg/tracing"
 	dtprinter "github.com/carolynvs/datetime-printer"
+
+	"reflect"
```

Comment:
Hi @troy0820 output of the porter installation list -o json gives output in this format

 ```json
[
  {
    "schemaType": "Installation",
    "schemaVersion": "1.0.2",
    "id": "01H8C1V1G19AH4PMQHAJ8PZBX9",
    "name": "porter-hello",
    "namespace": "",
    "bundle": {
      "repository": "ghcr.io/getporter/examples/porter-hello",
      "version": "0.2.0"
    },
    "status": {
      "runId": "01H8C1V4YEPKQK098NRFKD4AYQ",
      "action": "install",
      "resultId": "01H8C1VGHPZCEZRP3J4ATYCX3X",
      "resultStatus": "succeeded",
      "created": "2023-08-21T18:11:42.657996+05:30",
      "modified": "2023-08-21T18:11:46.125495+05:30",
      "installed": "2023-08-21T18:11:58.070842+05:30",
      "uninstalled": null,
      "bundleReference": "ghcr.io/getporter/examples/porter-hello:v0.2.0",
      "bundleVersion": "0.2.0",
      "bundleDigest": "sha256:276b44be3f478b4c8d1f99c1925386d45a878a853f22436ece5589f32e9df384"
    },
    "_calculated": {
      "resolvedParameters": null,
      "displayInstallationState": "installed",
      "displayInstallationStatus": "succeeded"
    }
  }
]
```

 The field names in the json are based on the json tags associated with the displayInstallation struct. Since the JSON output field names are what the users see in the output, in my opinion, those are ideally the selector fields which we should be using in the --field-selector. Reflect is being used to get the field values from the displayInstallation struct based on json tag based field selector values

Reply:
Marking this as resolved as per our discussion.